tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.h

```diff
@@ -102,8 +119,27 @@ struct OptimizationProfileConfig {
     }
     return true;
   }
+
+ private:
+  // In dynamic shape mode the optimization profile dims are only allowed to
+  // differ from the network input dims where the network input dims have -1
+  // values. We enforce this condition by changing prof_dims if necessary.
+  void enforceCompatibility(nvinfer1::Dims* prof_dims,
+                            const nvinfer1::Dims& input_dims) {
+    for (int i = 0; i < input_dims.nbDims; i++) {
+      if (input_dims.d[i] != -1) {
+        prof_dims->d[i] = input_dims.d[i];
+      }
+    }
+  }
 };
 
+// Optimization profile generation strategies.
+//
+// The kUnitTest option is only used for unit testing to avoid TRT bug that
+// removes redundant input shape values (200641519).
+enum class ProfileStrategy { kImplicitBatchLike, kMultiple, kUnitTest };
```

Comment:
Do these names come from TRT ? 
Because I think we can come up with a more "descriptive" naming.

- `kImplicitBatchLike` to me would be better named: `kLegacyImplicitBatchMode` because it's supposed to represent a Legacy Mode.

- `kMultiple` and `kUnitTest` by themselves don't cary any meaning, if I read them in the code, I have no idea what they represent.